kernel/api/src/main/java/org/sakaiproject/util/RequestFilter.java

```diff
@@ -1351,15 +1365,25 @@ protected Cookie findCookie(HttpServletRequest req, String name, String suffix)
 		{
 			for (int i = 0; i < cookies.length; i++)
 			{
-				if (cookies[i].getName().equals(name))
+				Cookie cookie = cookies[i];
```

Comment:
@jayee does this solve your issue:
```
			for (Cookie cookie : cookies)
			{
				if (StringUtils.equals(cookie.getName(), name))
				{
					if (StringUtils.endsWith(cookie.getValue(), StringUtils.join(DOT, suffix)))
					{
						return cookie;
					}
				}
			}
```

Reply:
Yes, it should work.
The important part about not returning a cookie when no suffix match in value is in place.

Reply:
Do you mind testing this first, to see if it addreses it.

Reply:
I'm no longer working with our sakai plattform, but I will ping those who do.